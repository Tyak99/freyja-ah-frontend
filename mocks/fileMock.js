<<<<<<< HEAD
export default '';
=======
module.exports = 'test-file-stub';
>>>>>>> ft(forgot-password-page): user can reset password page
