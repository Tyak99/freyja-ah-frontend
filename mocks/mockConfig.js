import axios from 'axios';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';

export const middlewares = [thunk];
export const mockStore = configureMockStore(middlewares);

export const mock = new MockAdapter(axios);
