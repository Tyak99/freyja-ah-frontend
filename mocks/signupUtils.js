/* eslint-disable import/prefer-default-export */
export const props = {
  auth: {
    isAuthenticated: true,
    loading: null,
    user: {
      userId: 21,
      email: 'harufffna@gmail.com',
      userName: 'ooche',
      iat: 1562518707,
      exp: 1563123507,
    },
  },
  errors: {
    isAuthenticated: false,
    loading: null,
    email: '',
    password: '',
    message: '',
  },
};
