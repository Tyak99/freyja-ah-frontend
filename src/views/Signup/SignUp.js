import React, { Fragment } from 'react';
import { Button, FormGroup, Input, Card, CardBody } from 'reactstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import classnames from 'classname';
import { registerUser } from '../../../store/actions/authActions';
import Logo from '../../assets/images/logo3.png';
import Footer from '../../components/footer/footer';
import Heading from '../../components/heading/heading';
import Navbar from '../../components/navbar/navbar';
import 'bootstrap/dist/css/bootstrap.css';
import './SignUp.scss';

class SignUp extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      userName: '',
      email: '',
      password: '',
      confirmPassword: '',
      errors: {},
    };
    this.onSubmit = this.onSubmit.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({
        errors: nextProps.errors.response.data,
      });
    }
  }

  onChange(e) {
    const { name } = e.target;
    const { value } = e.target;
    this.setState({ [name]: value });
  }

  onSubmit(e) {
    const {
      firstName,
      lastName,
      userName,
      email,
      password,
      confirmPassword,
    } = this.state;
    const { history } = this.props;
    e.preventDefault();
    const newUser = {
      firstName,
      lastName,
      userName,
      email,
      password,
      confirmPassword,
    };

    this.props.registerUser(newUser, history);
  }

  render() {
    const { errors } = this.state;
    // console.log(errors);
    let displayError;
    let secDisplayError;
    displayError = errors.error;
    secDisplayError = errors.error;
    if (errors && typeof secDisplayError === 'string') {
      secDisplayError = errors.error;
      console.log(secDisplayError);
    }

    if (errors && typeof secDisplayError === 'object') {
      displayError = errors.error;
      console.log(displayError);
    }

    return (
      <Fragment>
        <section className="sec-signup">
          <Navbar className="navbar" />
          <div className="container">
            <Card className="signup-box">
              <form className="sign-form" onSubmit={this.onSubmit}>
                <img src={Logo} alt="Logo" />
                <Heading title="CREATE AN ACCOUNT" />
                {secDisplayError && (
                  <div className="feedbackk">{secDisplayError}</div>
                )}
                <FormGroup className="inf">
                  <Input
                    className={classnames('input firstname', {
                      'is-invalid': displayError,
                    })}
                    type="text"
                    placeholder="First Name"
                    name="firstName"
                    value={this.state.firstName}
                    onChange={e => this.onChange(e)}
                  />
                  {displayError && (
                    <div className="feedback">{displayError[0].msg}</div>
                  )}
                </FormGroup>
                <FormGroup className="inf">
                  <Input
                    className={classnames('input lastname', {
                      'is-invalid': displayError,
                    })}
                    type="text"
                    placeholder="Last Name"
                    name="lastName"
                    value={this.state.lastName}
                    onChange={e => this.onChange(e)}
                  />
                  {displayError && (
                    <div className="feedback">{displayError[2].msg}</div>
                  )}
                </FormGroup>
                <FormGroup className="inf">
                  <Input
                    className="input lastname"
                    type="text"
                    placeholder="User Name"
                    name="userName"
                    value={this.state.userName}
                    onChange={e => this.onChange(e)}
                  />
                </FormGroup>
                <FormGroup className="inf">
                  <Input
                    className={classnames('input email', {
                      'is-invalid': displayError,
                    })}
                    type="email"
                    placeholder="Email"
                    name="email"
                    value={this.state.email}
                    onChange={e => this.onChange(e)}
                  />
                  {displayError && (
                    <div className="feedback">{displayError[4].msg}</div>
                  )}
                </FormGroup>
                <FormGroup className="inf">
                  <Input
                    className={classnames('input password', {
                      'is-invalid': displayError,
                    })}
                    type="password"
                    placeholder="Password"
                    name="password"
                    value={this.state.password}
                    onChange={e => this.onChange(e)}
                  />
                  {displayError && (
                    <div className="feedback">{displayError[5].msg}</div>
                  )}
                </FormGroup>
                <FormGroup className="inf">
                  <Input
                    className={classnames('input cpassword', {
                      'is-invalid': displayError,
                    })}
                    type="password"
                    placeholder="Confirm Password"
                    name="confirmPassword"
                    value={this.state.confirmPassword}
                    onChange={e => this.onChange(e)}
                  />
                  {displayError && (
                    <div className="feedback">{displayError[6].msg}</div>
                  )}
                </FormGroup>
                <Button className="btnSubmit">Sign up</Button>{' '}
              </form>
              <CardBody />
            </Card>
          </div>
        </section>
        <Footer />
      </Fragment>
    );
  }
}
SignUp.propTypes = {
  registerUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
};
const mapStateToProps = state => ({
  auth: state.auth,
  errors: state.errors,
});
export default connect(
  mapStateToProps,
  { registerUser },
)(withRouter(SignUp));
