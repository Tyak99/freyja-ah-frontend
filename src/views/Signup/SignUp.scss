html {
  color: #6d6666;
}
.sec-signup {
  width: 100%;
  background-image: linear-gradient(rgba(5, 0, 0, 0.7), rgba(0, 0, 0, 0.7)),
    url('../../assets/images/signupbg.jpg');
  background-size: cover;
  margin: auto;
}
.container {
  width: 1200px;
  padding: 0 15px;
  margin: auto;
  padding-top: 3%;
  padding-bottom: 3%;
}
.signup-box {
  width: 400px;
  background: #ffffff;
  border-radius: 5px;
  margin: auto;
  z-index: 1000;
  text-align: center;
  padding-top: 2%;

  @media screen and (max-width: 420px) {
    width: 300px;
  }
}

.navbar {
  background-color: #5f3518 !important;
}
.btnSubmit,
.firstname,
.lastname,
.email,
.password,
.cpassword {
  border: 0.5px solid #707070;
  width: 90%;
  padding: 6%;
  border-radius: 5px;
}
.boxx-tex {
  margin-bottom: 3%;
}
.inf {
  margin-bottom: 1.5rem;
}
.btnSubmit,
.input {
  margin: auto;
}
.btnSubmit {
  background-color: #f2994a;
  border: 0.5px solid #f2994a;
  padding: 3.5%;
}
.form-control.is-invalid {
  background-image: none !important;
}
.btnSubmit:hover,
.btnSubmit:active,
.btnSubmit:focus {
  background-color: #f2994a;
  border: 0.5px solid #f2994a;
}
.btnSubmit:hover {
  opacity: 0.6;
}
.logo {
  margin-top: 2%;
}
.feedbackk,
.feedback {
  color: red;
  padding: 0.3em 0 0 0.4em;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 14px;
  text-align: left;
  margin-left: 5%;
}
.feedbackk {
  text-align: center;
  margin-bottom: 2%;
}
.headsi {
  margin-top: 2%;
  margin-bottom: 3%;
}
