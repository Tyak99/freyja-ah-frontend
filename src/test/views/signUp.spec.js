import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
// import store from '../../../store/index';
import Signup from '../../views/Signup/SignUp';
import { mockStore } from '../../../mocks/mockConfig';

const newUser = {
  firstName: 'Jones',
  lastName: 'Allen',
  userName: 'Allen',
  email: 'jonesallen@gmail.com',
  password: 'password',
  confirmPassword: 'password',
};
const store = mockStore({ user: { newUser } });
const setUp = () => {
  const component = shallow(<Signup store={store} />);
  return component;
};

describe('Signup component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it('should render without errors', () => {
    expect(component).toMatchSnapshot();
  });
});
