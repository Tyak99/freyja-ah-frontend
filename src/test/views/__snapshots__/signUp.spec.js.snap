// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component should render without errors 1`] = `ShallowWrapper {}`;
